fang/src/asynk/backend_sqlx/mysql.rs

```diff
@@ -0,0 +1,351 @@
+                        .await?;
+                Ok(Res::Task(task))
+            }
+
+            Q::FailTask => {
+                let task = mysql_impl_fail_task(FAIL_TASK_QUERY_MYSQL, pool, params).await?;
+
+                Ok(Res::Task(task))
+            }
+
+            Q::RemoveAllTask => {
+                let affected_rows =
+                    general_any_impl_remove_all_task(REMOVE_ALL_TASK_QUERY_MYSQL, pool).await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+
+            Q::RemoveAllScheduledTask => {
+                let affected_rows = general_any_impl_remove_all_scheduled_tasks(
+                    REMOVE_ALL_SCHEDULED_TASK_QUERY_MYSQL,
+                    pool,
+                )
+                .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+
+            Q::RemoveTask => {
+                let affected_rows =
+                    general_any_impl_remove_task(REMOVE_TASK_QUERY_MYSQL, pool, params).await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::RemoveTaskByMetadata => {
+                let affected_rows = general_any_impl_remove_task_by_metadata(
+                    REMOVE_TASK_BY_METADATA_QUERY_MYSQL,
+                    pool,
+                    params,
+                )
+                .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::RemoveTaskType => {
+                let affected_rows =
+                    general_any_impl_remove_task_type(REMOVE_TASKS_TYPE_QUERY_MYSQL, pool, params)
+                        .await?;
+
+                Ok(Res::Bigint(affected_rows))
+            }
+            Q::FetchTaskType => {
+                let task =
+                    general_any_impl_fetch_task_type(FETCH_TASK_TYPE_QUERY_MYSQL, pool, params)
+                        .await?;
+                Ok(Res::Task(task))
+            }
+            Q::FindTaskById => {
+                let task: Task =
+                    general_any_impl_find_task_by_id(FIND_TASK_BY_ID_QUERY_MYSQL, pool, params)
+                        .await?;
+
+                Ok(Res::Task(task))
+            }
+            Q::RetryTask => {
+                let task = mysql_impl_retry_task(RETRY_TASK_QUERY_MYSQL, pool, params).await?;
+
+                Ok(Res::Task(task))
+            }
+            Q::InsertTaskIfNotExists => {
+                let task = mysql_any_impl_insert_task_if_not_exists(
+                    (
+                        FIND_TASK_BY_UNIQ_HASH_QUERY_MYSQL,
+                        INSERT_TASK_UNIQ_QUERY_MYSQL,
+                    ),
+                    pool,
+                    params,
+                )
+                .await?;
+
+                Ok(Res::Task(task))
+            }
+        }
+    }
+
+    pub(super) fn _name() -> &'static str {
+        "MySQL"
+    }
+}
+
+async fn mysql_impl_insert_task(
```

Comment:
what do you think about defining these functions as part of the trait?

so mysql, postgres and sqlite will be implementing this trait

Reply:
This will be explored #148 in this issue.